data-generation/src/main/scala/benchmarks/QueryBenchmark.scala

```diff
@@ -0,0 +1,27 @@
+package benchmarks
+
+import org.apache.spark.SparkContext
+import thesis.SparkConfiguration.getSparkSession
+import utils.{Benchmark, FileWriter}
+
+abstract class QueryBenchmark(val iterationCount: Int, val customColumn: String) extends Serializable {
+  implicit val sc: SparkContext = getSparkSession.sparkContext
+  var benchmark: Benchmark = null
+
+  def run: Unit = {
+    initialize()
+    for (i <- 1 to iterationCount) {
+      execute(i)
+    }
+    tearDown()
+  }
+
+  def initialize(): Unit = {
+    val fileWriter = FileWriter(filename = getClass.getSimpleName)
+    benchmark = Benchmark(fileWriter, textPrefix = getClass.getSimpleName, customColumn = customColumn)
```

Comment:
Så nå får ikke benchmarkesene sette navnet på benchmarken selv?
Eller blir navnet satt til den klassen som extender QueryBenchmark?

Reply:
> Eller blir navnet satt til den klassen som extender QueryBenchmakr?

Stemmer. Tenker i utgangspunktet at det gir kontekst nok for filnavn